providers/implementations/asymciphers/rsa_enc.c

```diff
@@ -425,7 +425,7 @@ static int rsa_set_ctx_params(void *vprsactx, const OSSL_PARAM params[])
     const OSSL_PARAM *p;
     char mdname[OSSL_MAX_NAME_SIZE];
     char mdprops[OSSL_MAX_PROPQUERY_SIZE] = { '\0' };
-    char *str = mdname;
+    char *str;
```

Comment:
I'd better initialized it to NULL here

Reply:
I considered initializing `str`, but noticed that `p` is also not initialized.

Why it is better to initialize `str` to NULL?

Reply:
IMHO as you will set it to NULL in case you didn't find any suitable parameters, it's worth doing it from the beginning. It also will ensure that there is no paths where this value stays initialized with some garbage. Modern analyzers catch these cases but for me it looks better to ensure it explicitly.

Reply:
ok.  thanks for your explanation.

I've added the initializer.